lib/crawlerClient.js

```diff
@@ -62,19 +62,25 @@ class CrawlerClient {
     return this._call('get', `${this.url}/deadletters`, `Failed to list deadletters`);
   }
 
+  getDeadletterCount() {
+    return this._call('head', `${this.url}/deadletters`, `Failed to list deadletters`, null, 'response').then(response => {
+      return parseInt(response.headers['x-total-count'], 10);
+    });
+  }
+
   getDeadletter(urn) {
-    return this._call('get', `${this.url}/deadletters/${encodeURIComponent(urn)}`, `Failed to get deadletter`);
+    return this._call('get', `${this.url}/deadletters/${encodeURIComponent(urn)}`, `Failed to get deadletter ${urn}`);
   }
 
   deleteDeadletter(urn) {
-    return this._call('delete', `${this.url}/deadletters/${encodeURIComponent(urn)}`, `Failed to delete deadletter`);
+    return this._call('delete', `${this.url}/deadletters/${encodeURIComponent(urn)}`, `Failed to delete deadletter ${urn}`);
   }
 
-  requeueDeadletter(urn, queue) {
-    return this._call('delete', `${this.url}/deadletters/${encodeURIComponent(urn)}?requeue=${encodeURIComponent(queue)}`, `Failed to get deadletter`);
+  requeueDeadletter(urn, queue = 'soon') {
+    return this._call('delete', `${this.url}/deadletters/${encodeURIComponent(urn)}?requeue=${encodeURIComponent(queue)}`, `Failed to requeue deadletter ${urn} on ${queue}`);
   }
 
-  _call(method, url, errorMessage, body = null) {
+  _call(method, url, errorMessage, body = null, returnType = 'body') {
```

Comment:
I had considered a different way of doing this, which was in the body if the method is 'head' then return the X-Total-Count header instead of the body (basically codifying that we use HEAD to get counts). Your implementation is fine just wanted to share this option 😄